io/input/router.go

```diff
@@ -567,6 +567,14 @@ func (q *Router) changeState(e event.Event, state inputState, evts []taggedEvent
 			e.event = de
 		}
 	}
+	for i := range evts {
+		e := &evts[i]
+		if fe, ok := e.event.(key.FocusEvent); ok {
+			if !fe.Focus {
+				state.keyState.focus = nil
```

Comment:
It seems to me this should be done in `Router.processEvent`, near the lines

```go
	case key.EditEvent, key.FocusEvent, key.SelectionEvent:
		var evts []taggedEvent
		if f := state.focus; f != nil {
			evts = append(evts, taggedEvent{tag: f, event: e})
		}
		q.changeState(e, state, evts)
```

But I'm not perfectly sure what use-case you're fixing. Can you add a test to io/input/key_test.go for my understanding and to guard against regressions?

Reply:
What I'm fixing is reseting the `focus` when receiving `key.FocusEvent` from "the OS".

Consider that you have one `FocusCmd{Tag: TextArea}`. If Gio gets one `FocusEvent` from "the OS" (say: macos), it will not remove the focus from `TextArea`. That is the issue here. 

So, this change changes the `state` (state.keyState.focus`) to `nil` when the FocusEvent is `false`. Since it's messing with `state` I use the `changeState` function.